types/types.go

```diff
@@ -302,6 +306,58 @@ type Alert struct {
 	Timeout   bool
 }
 
+var validateLs = func(ls model.LabelSet) error {
```

Comment:
I think we can do a bit better here - I'd hope we can do dependency injection for these changes.

1. It's future-proofing ourselves because as we add more feature flags, it's very likely we'd need to control these components.
2. It'd make things easier to follow (e.g. you wouldn't need to override the call to `Validate` or have a need to use global variables.